Feature/simplifiy localization form

JevgenijVisockij opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplfies localization form
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  | Simplifying Improve -> International -> Localization form. Everything must look and work mostly the same. Main changes are replacement of blue help box by help text under input. Another change is that there is now a red * next to some type as they are actually required fields. 

:notebook: **BC Break**
Backwards compatibility break introduced due to LocalUnitsType, AdvancedConfigurationType and LocalizationConfigurationType extending of TranslationAwareType. This means if any module extends any of those files they will get an exception.
<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/21243)
<!-- Reviewable:end -->

JevgenijVisockij committed:
CS Fixer fix

JevgenijVisockij committed:
Removed accidentally added code

JevgenijVisockij committed:
Made all local unit types required again

JevgenijVisockij committed:
Removed required => false where they are not needed

JevgenijVisockij committed:
Fixed some comments not being correct

matks reviewed (dismissed):
Great PR

ngodefroy commented:
Hello @JevgenijVisockij 
What I tested:
Non Regression from develop version
Checked the red cross & text under input is applied.
<img width="980" alt="Capture d’écran 2020-10-12 à 14 31 22" src="https://user-images.githubusercontent.com/69145330/95746741-9ab3c600-0c97-11eb-989d-778b824c5aa6.png">
==>**TEST OK**

Thank you

matks commented:
@JevgenijVisockij unfortunately we need a rebase for this PR 😢

JevgenijVisockij committed:
Simplified Localization form

# Conflicts:
#	src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

matks commented:
Thank you @JevgenijVisockij
